vaadin-time-picker-flow/src/main/java/com/vaadin/flow/component/timepicker/TimePicker.java

```diff
@@ -395,52 +395,118 @@ public Locale getLocale() {
+     *            <code>null</code> to remove any minimum constraints
+     */
+    @Override
+    public void setMin(LocalTime min) {
+        this.min = min;
+        super.setMin(min);
+    }
+
     /**
      * Gets the minimum time in the time picker. Time before that will be
      * disabled in the popup.
      *
+     * @deprecated use {@link #getMinTime()} instead.
+     *
      * @return the minimum time that is allowed to be selected, or
      *         <code>null</code> if there's no minimum
      */
+    @Deprecated
     public String getMin() {
         return super.getMinString();
     }
 
+    /**
+     * Gets the minimum time in the time picker. Time before that will be
+     * disabled in the popup.
+     *
+     * @return the minimum time that is allowed to be selected, or
+     *         <code>null</code> if there's no minimum
+     */
+    public LocalTime getMinTime() {
+        String minString = super.getMinString();
+        return minString.isEmpty() ? null :
+                LocalTime.parse(super.getMinString(), initializeAndReturnFormatter());
+    }
+
     /**
      * Sets the maximum time in the time picker. Times after that will be
      * disabled in the popup.
      *
+     * @deprecated use {@link #setMax(LocalTime)} instead.
+     *
      * @param max
      *            the maximum time that is allowed to be selected, or
      *            <code>null</code> to remove any maximum constraints
      */
     @Override
+    @Deprecated
     public void setMax(String max) {
         this.max = LocalTime.parse(max, initializeAndReturnFormatter());
         super.setMax(max);
     }
 
+    /**
+     * Sets the maximum time in the time picker. Times after that will be
+     * disabled in the popup.
+     *
+     * @param max
+     *            the maximum time that is allowed to be selected, or
+     *            <code>null</code> to remove any maximum constraints
+     */
+    @Override
+    public void setMax(LocalTime max) {
+        this.max = max;
+        super.setMax(max);
+    }
+
     /**
      * Gets the maximum time in the time picker. Times after that will be
      * disabled in the popup.
      *
+     * @deprecated use {@link #getMaxTime()} instead.
+     *
      * @return the maximum time that is allowed to be selected, or
      *         <code>null</code> if there's no maximum
      */
+    @Deprecated
     public String getMax() {
         return super.getMaxString();
     }
 
+    /**
+     * Gets the maximum time in the time picker. Times after that will be
+     * disabled in the popup.
+     *
+     * @return the maximum time that is allowed to be selected, or
+     *         <code>null</code> if there's no maximum
+     */
+    public LocalTime getMaxTime() {
+        String maxString = super.getMaxString();
+        return maxString.isEmpty() ? null :
+                LocalTime.parse(super.getMaxString(), initializeAndReturnFormatter());
```

Comment:
The setter uses always `ISO_LOCAL_TIME` to format the time, but getter uses the user-defined locale. That doesn't seem right.

But actually, this could just return `this.max`.